pkg/api/bzzchunk.go

```diff
@@ -33,6 +39,18 @@ func (s *server) chunkUploadHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// create the Tag for the chunk
+	tagName := r.Header.Get(TagHeaderName)
+	if tagName == "" {
+		tagName = fmt.Sprintf("unnamed_tag_%d", time.Now().Unix())
+	}
+	tag, err := s.Tags.Create(tagName, 1, false)
```

Comment:
this is a bit tricky and we might need to change the API here. here's why:
- right now tags.Create creates a random number for a tag
- this means that if you need to upload several chunks that correlate to the same upload you cannot do so, since even if you provide the same tag name, you will get a new tag for every new chunk you upload, so you cannot keep track and associate multiple chunks with the same tag
- this might work for an API that accepts arbitrary length input, but not for a single chunk API

off the top of my head, i can think about 2 different possibilities of calling the single chunk api:
1. first chunk in a series of correlating chunks, we still don't have a tag at this point so: we call the api with the `x-swarm-tag` header, and create a new tag, this tag ID that was created should be returned in a header or the response somehow
2. chunk # 2 or more in the above sequence, we now have a tag ID and we would like to associate this chunk with the tag, so we could `POST` to the api with a separate `x-swarm-tag-id` header containing the actual tag, then in the handler we can do the appropriate increments to the tag. `x-swarm-tag` header in this case _should_ be ignored and is unnecessary since the tag already exists

WDYT about this? does this make sense?

Reply:
First of all ..this is meant for only single chunks, not a series of chunks that needs to be tagged ( i mean the  scope of the PR). If someone wants to send a series of chunks and tag them with same id or something... i would check if that hash a UUID already and not create tag, instead pull the tag from store.


- From the external application point of view, it should send first chunk, get the response, pick the UUID and use that in every subsequent chunks.

Reply:
well currently the `id` is just a string meant to identify the entire chain of chunks in a tag (in old swarm codebase, the functionality covers multiple chunks within one tag). so it was created in the sense that you could tag an upload like `"upload of directory ./documents"` (or some other general string identifier used to identify the tag from another L2 application that uses swarm as of such).
So but now you're saying that the `id` could also be a general string or a uuid? not sure if I'm following

Reply:
please note that right now the general `id` string is not enforcing any unique constraints in `tags`. I think mixing these things up using the same field will create confusion and possible bugs

Reply:
@acud I am no0t sure i follow you. I dont see any issues. now you have one chunk one tag because we can uppload one chunk unlike previous swarm code.

Reply:
This question is on what functionality we want to have with chunk tags. As @acud, I also had the impression that we would like to be able have chunks grouped by the common tag for them so that their progress can be followed. We would need to increment Total in Tag when a new chunk is uploaded with the same tag name.

Reply:
@janos tag name or tag ID

Reply:
It should be able to be identified by header `x-swarm-tag`, so I suppose: tag name.

Reply:
i changed it to x-swarm-tag-uid to be specific.  It was very confusing as we have a name variable also in tag, but that is just for name sake :-)

Reply:
Thanks, that is fine.